cloud/terraform/azure_config_builder_v2.py

```diff
@@ -0,0 +1,332 @@
+
+    def __new_azure_subnet(self, instance):
+        name = self.create_resource_name([instance['location'], 'subnet'])
+        instance['azurerm_subnet'] = name
+
+        new_subnet = {
+            'name': name,
+            'resource_group_name': self.resource_group,
+            'virtual_network_name': instance['azurerm_virtual_network'],
+            'address_prefixes': ['10.0.2.0/24'],
+            'depends_on': [
+                'azurerm_virtual_network.{}'.format(instance['azurerm_virtual_network']),
+            ]
+        }
+
+        self.resources_tf['resource']['azurerm_subnet'][name] = new_subnet
+
+    def __new_azure_public_ip(self, instance):
+        name = self.create_resource_name([instance['hostname'], 'public-ip'])
+        instance['azurerm_public_ip'] = name
+
+        new_public_ip = {
+            'name': name,
+            'resource_group_name': self.resource_group,
+            'location': instance['location'],
+            'allocation_method': 'Static',
+            'domain_name_label': instance['hostname'],
+            'tags': {self.ci_tag_key: self.ci_test_value}
+        }
+
+        self.resources_tf['resource']['azurerm_public_ip'][name] = new_public_ip
+
+    def __new_azure_nic(self, instance):
+        name = self.create_resource_name([instance['hostname'], 'nic'])
+        instance['azurerm_network_interface'] = name
+
+        ip_configuration = {
+            'name': self.create_resource_name(['ip-config']),
+            'subnet_id': self.__get_azure_network_resource_uri(
+                terraform_resource_type='azurerm_subnet',
+                azure_resource_name=instance['azurerm_subnet'],
+                azure_virtual_network_name=instance['azurerm_virtual_network']),
+            'private_ip_address_allocation': 'Dynamic',
+            'public_ip_address_id': self.__get_azure_network_resource_uri(
+                terraform_resource_type='azurerm_public_ip',
+                azure_resource_name=instance['azurerm_public_ip'])
+        }
+
+        new_nic = {
+            'name': name,
+            'location': instance['location'],
+            'resource_group_name': self.resource_group,
+            'ip_configuration': ip_configuration,
+            'tags': {self.ci_tag_key: self.ci_test_value},
+            'depends_on': [
+                'azurerm_virtual_network.{}'.format(instance['azurerm_virtual_network']),
+                'azurerm_subnet.{}'.format(instance['azurerm_subnet']),
+                'azurerm_public_ip.{}'.format(instance['azurerm_public_ip']),
+            ]
+        }
+
+        self.resources_tf['resource']['azurerm_network_interface'][name] = new_nic
+
+    def __new_azure_vm(self, instance):
+        if 'instance_type' not in instance or not instance['instance_type']:
+            if instance['arch'] == 'x86_64':
+                instance['instance_type'] = self.default_x86_vm_size
+            elif instance['arch'] == 'arm64':
```

Comment:
I would add "else" here. Sometimes we don't specify the VM size and the default one should be used (assuming that it's x64 arch - we do the same for AWS).